### ISSUE TITLE
practical translate. Help needed

### ISSUE THREAD

**Issue by remotejob** [graviraja/seq2seq#1]
Can you add real translate function for 'transformer' as for 'padded_seq2seq.py'

`def translate_sentence(sentence):
    # tokenize the sentence
    tokenized = tokenize_de(sentence)
    tokenized = ['<sos>'] + [t.lower() for t in tokenized] + ['<eos>']
`

Thanks.

**Comment by whoszus** [comment]
I have the same issue with @remotejob, that I need a function to implement the translation and to get accuracy. 
Thanks.

**Comment by whoszus** [comment]
I read a sentence on your blog that you said that
 'We will send the trg sentence and the encoded src sentence to the decoder, to get the target sentence.' 
What's it means?

**Comment by graviraja** [comment]
Hi,

It means that in the case of training, the decoder of transformer takes the
encoded src sentence, and the target sentence words as inputs in order to
predict the sentence.
Where as in the case of evaluation/inference, decoder takes the encoded src
sentence, and the predicted word as input.

In simple terms,
For training we will use the target sentence as input to decoder, in order
to predict the target sentence and calculate the loss on how much different
is the predicted from target sentence.
In evaluation, we will use the predicted word at time step t as the input
to time step t+1. We use target sentence to calculate the loss.

Hope this helps.

Thanks,
Ravi.

**Comment by whoszus** [comment]
Great,   Thx.

**Comment by whoszus** [comment]
And I found  another implementation of this paper, hope this helps @remotejob 
https://github.com/jadore801120/attention-is-all-you-need-pytorch
